src/executor/source/executor/nodes/ScriptNode.ts

```diff
@@ -19,6 +19,12 @@ export async function executeScriptNode( node: Node<ScriptNode>, input: Record<s
     sandbox: { inputs: Object.values( input ) }
   } )
   const result = vm.run( `${node.data.params.function}\nhandler(...inputs)` )
-
+  if ( node.data.outgoingHandles.length > 0 ){
+    const results = {}
+    for ( const outgoingHandle of node.data.outgoingHandles ){
+      Object.assign( results, { [outgoingHandle.id]: result } )
+    }
+    return Promise.resolve( results )
+  }
```

Comment:
See above.

Also: You can just return the object, `Promise.resolve` is redundant.